src/components/sections/ProductGallery/ProductGallery.tsx

```diff
@@ -31,100 +32,150 @@ function ProductGallery({ title }: Props) {
+                >
+                  Filters
+                </Button>
+              )}
+            </div>
+          </>
         )}
-      </div>
 
-      {/* Controls */}
-      <div>
-        <div data-testid="total-product-count" data-count={totalCount}>
-          Total Products: {totalCount}
+        <div className="product-listing__filters-bar">
+          <Suspense fallback={null}>
+            <Filter
+              isOpen={isFilterOpen}
+              facets={orderedFacets}
+              onDismiss={() => setIsFilterOpen(false)}
+            />
+          </Suspense>
         </div>
-        <Sort />
-        <h2>Most Wanted</h2>
-      </div>
 
-      {/* Add link to previous page. This helps on SEO */}
-      {prev !== false && (
-        <>
-          <GatsbySeo linkTags={[{ rel: 'prev', href: prev.link }]} />
-          <a
-            onClick={(e) => {
-              e.currentTarget.blur()
-              e.preventDefault()
-              addPrevPage()
-            }}
-            href={prev.link}
-            rel="prev"
-          >
-            Previous Page
-          </a>
-        </>
-      )}
-
-      {/* Render ALL products */}
-      {pages.map((page) => (
-        <GalleryPage
-          key={`gallery-page-${page}`}
-          fallbackData={page === searchState.page ? data : undefined}
-          page={page}
-          title={title}
-        />
-      ))}
-
-      {/* Add link to next page. This helps on SEO */}
-      {next !== false && (
-        <>
-          <GatsbySeo linkTags={[{ rel: 'next', href: next.link }]} />
-          <LinkButton
-            data-testid="show-more"
-            onClick={(e) => {
-              e.currentTarget.blur()
-              e.preventDefault()
-              addNextPage()
-            }}
-            href={next.link}
-            rel="next"
-            variant="secondary"
-          >
-            Load more products
-          </LinkButton>
-        </>
-      )}
-
-      {/* Prefetch Previous and Next pages */}
-      {prev !== false && (
-        <GalleryPage page={prev.cursor} display={false} title={title} />
-      )}
-      {next !== false && (
-        <GalleryPage page={next.cursor} display={false} title={title} />
-      )}
-    </>
+        {data ? (
+          <>
+            <div className="product-listing__results">
+              {/* Add link to previous page. This helps on SEO */}
+              {prev !== false && (
+                <>
+                  <GatsbySeo linkTags={[{ rel: 'prev', href: prev.link }]} />
+                  <a
+                    onClick={(e) => {
+                      e.currentTarget.blur()
+                      e.preventDefault()
+                      addPrevPage()
+                    }}
+                    href={prev.link}
+                    rel="prev"
+                  >
+                    Previous Page
+                  </a>
+                </>
+              )}
```

Comment:
I think we need to handle this Previous Page afterward in another task 👀 
<img width="1287" alt="Screen Shot 2022-01-26 at 20 00 33" src="https://user-images.githubusercontent.com/11325562/151261699-f56ae13d-4382-4222-b01f-b3c5eda1ef3e.png">

How to show this link?
1. Hit the Load More products button.
2. Scroll to the next page.
3. Either refresh the page or go to PDP and come back using the browser back history button.

Reply:
We decided to style this link as a button, just like the "Load more products" one. Done it [here](https://github.com/vtex-sites/base.store/commit/f8b14a66d42a109c8409a41ccee87fae74e46945).